src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts

```diff
@@ -42,34 +42,61 @@ export class DotPushPublishContentTypesDialogComponent implements OnInit, OnDest
-                'contenttypes.content.push_publish.action.pushremove',
-                'contenttypes.content.push_publish.I_want_To',
-                'contenttypes.content.push_publish.force_push',
-                'contenttypes.content.push_publish.publish_date',
-                'contenttypes.content.push_publish.expire_date',
-                'contenttypes.content.push_publish.push_to',
-                'contenttypes.content.push_publish.push_to_errormsg',
-                'contenttypes.content.push_publish.form.cancel',
-                'contenttypes.content.push_publish.form.push',
-                'contenttypes.content.push_publish.publish_date_errormsg',
-                'contenttypes.content.push_publish.expire_date_errormsg'
-            ])
+        const messages$ = this.dotMessageService.getMessages([
+            'contenttypes.content.push_publish',
+            'contenttypes.content.push_publish.filters',
+            'contenttypes.content.push_publish.action.push',
+            'contenttypes.content.push_publish.action.remove',
+            'contenttypes.content.push_publish.action.pushremove',
+            'contenttypes.content.push_publish.I_want_To',
+            'contenttypes.content.push_publish.force_push',
+            'contenttypes.content.push_publish.publish_date',
+            'contenttypes.content.push_publish.expire_date',
+            'contenttypes.content.push_publish.push_to',
+            'contenttypes.content.push_publish.push_to_errormsg',
+            'contenttypes.content.push_publish.form.cancel',
+            'contenttypes.content.push_publish.form.push',
+            'contenttypes.content.push_publish.publish_date_errormsg',
+            'contenttypes.content.push_publish.expire_date_errormsg'
+        ]);
+        const filterOptions$ = this.dotPushPublishFiltersService.get();
+
+        combineLatest(messages$, filterOptions$)
             .pipe(takeUntil(this.destroy$))
-            .subscribe((messages: { [key: string]: string }) => {
-                this.pushActions = this.getPushPublishActions(messages);
-                this.initForm();
-                this.setDialogConfig(messages, this.form);
-                this.dialogShow = true;
-            });
+            .subscribe(
+                ([messages, filterOptions]: [
+                    { [key: string]: string },
+                    DotPushPublishFilter[]
+                ]) => {
+                    this.filterOptions = filterOptions.map((filter: DotPushPublishFilter) => {
+                        return {
+                            label: filter.title,
+                            value: filter.key
+                        };
+                    });
+
+                    const defaultFilterKey = filterOptions
+                        .filter((filter: DotPushPublishFilter) => {
+                            if (filter.default) {
+                                return filter;
```

Comment:
what happend here if filter.default is false?

why not just:
```
return filter.default
```

return filter.default;

Reply:
Is not what I'm trying to do, I'm getting the object that have `default: true`